### PR TITLE
Extract the color palette to its own global styles screen

### PR DESCRIPTION
**Pull request by youknowriad**
Related #34574 

This PR extracts the color palette to its own screen in the nested hierarchy of the global styles sidebar to match the designs in #34574.

**Testing instructions**

 - Try adding/removing colors from the global styles panel globally and for a specific block.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar/global-styles/style.scss
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar/global-styles/style.scss:21]: We keep duplicating these styles for the color indicator component, it seems it needs a `isRound` and a `size` prop maybe. Potentially make all indicators round.

File: packages/edit-site/src/components/sidebar/global-styles/palette.js
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar/global-styles/palette.js:9]: It feels like there's too many experimental UI components now, we should start making decisions about stability soon.

**Comment by jasmussen** [comment]
This is advancing so fast, incredible work Riad. At some point I'd like to offer to help you massage some of the details (paddings, margins, colors, metrics), but at this point I don't want to mess with the path you're on. The sliding mechanism and even the color stacks are looking great, but I do encounter a crash saving a particular color:
gs

This is the error I got:

Screenshot 2021-09-24 at 13 39 44

In case it's a bug in the theme I tested with, [you can grab that here](https://github.com/jasmussen/mocco).

**Comment by jasmussen** [comment]
On a separate note, the GIF above also shows how the sibling inserter sometimes lingers, and in doing so causes a horizontal scrollbar when for example a sidebar panel is toggled on. @ellatrix I think you worked on the sibling inserter, would this be a tricky fix? I'd be happy to open a separate ticket.

File: packages/edit-site/src/components/sidebar/global-styles/palette.js
Comment (mtias) [reply packages/edit-site/src/components/sidebar/global-styles/palette.js:9]: What do you have in mind? Is this the first place we are using them heavily? Maybe once we are done with the design tools and it all fits we can stabilize.

File: packages/edit-site/src/components/sidebar/global-styles/style.scss
Comment (mtias) [reply packages/edit-site/src/components/sidebar/global-styles/style.scss:21]: All color indicators are circular now, no?
cc @ciampo

Comment (youknowriad) [reply packages/edit-site/src/components/sidebar/global-styles/style.scss:21]: That component is an old one that is used for the old color panels (when closed they show a rectangle with a color), it's the only component that is exposed AFAIK. It feels we should just update it to be circular.

File: packages/edit-site/src/components/sidebar/global-styles/palette.js
Comment (youknowriad) [reply packages/edit-site/src/components/sidebar/global-styles/palette.js:9]: > Maybe once we are done with the design tools and it all fits we can stabilize.

Yes, that makes sense. It will allow us to assert that they are good enough before making the switch. I think some of them are fairly safe like "ZStack, HStack, VStack" though maybe we can be more explicit `HorizontaStack...`

File: packages/edit-site/src/components/sidebar/global-styles/style.scss
Comment (mtias) [reply packages/edit-site/src/components/sidebar/global-styles/style.scss:21]: That's my inclination as well

Comment (ciampo) [reply packages/edit-site/src/components/sidebar/global-styles/style.scss:21]: I've been taking a look at these inconsistencies and discussed them with a few folks (including @mtias) — I reported the outcome of that conversation in https://github.com/WordPress/gutenberg/issues/33820#issuecomment-904497769.

The TL;DR was that we agreed to look into creating a new component, called `ColorSwatch`, that would potentially replace both `ColorIndicator` and `CircularOptionPicker`, offering a better API and better a11y (borrowing from the g2 `ColorCircle`). It will be then used as the base for rewriting/improving higher-level components for managing color palettes.

I'm currently in the process of creating a tracking issue #35093 for color-related components which will includes the tasks related to `ColorSwatch` & co.

File: packages/edit-site/src/components/sidebar/global-styles/palette.js
Comment (ciampo) [reply packages/edit-site/src/components/sidebar/global-styles/palette.js:9]: I agree with Matias. Let's first use them in a real-world application before deciding if they're stable enough to be marked as non-experimental, especially given how much we want to avoid introducing breaking changes.

File: packages/edit-site/src/components/sidebar/global-styles/style.scss
Comment (mtias) [reply packages/edit-site/src/components/sidebar/global-styles/style.scss:21]: Thanks for the refresher — if it's all tracked we are good.

**Comment by youknowriad** [comment]
@jasmussen I fixed the error good catch.

For the spacing, I agree, I'd love for us to use UI components directly and not have to tweak spacing at all. I didn't explore this yet because we're in this intermediary step where we have some redesigned panels but not all.

**Comment by jasmussen** [comment]
> I didn't explore this yet because we're in this intermediary step where we have some redesigned panels but not all.

Yep, I only wanted to say: you know where to find me if/when you can use my help!

**Comment by youknowriad** [comment]
Could use a ✅ if you consider this good enough as an iteration :)

File: packages/components/src/item-group/test/__snapshots__/index.js.snap
Comment (youknowriad) [inline packages/components/src/item-group/test/__snapshots__/index.js.snap:15]: It's weird that I had to update the snapshots because I added a "margin" to the styles. I wonder if there's a way to avoid snapshots for unit testing components and make them not dependent on styles.

Comment (ciampo) [reply packages/components/src/item-group/test/__snapshots__/index.js.snap:15]: > make them not dependent on styles

I'd argue that the way a UI component is styled contributes to it functioning as expected. Using Emotion, it means that if the styles change, the generated classnames change as well.

> I wonder if there's a way to avoid snapshots for unit testing components

The way would be to delete those tests and replace them with more targeted tests.

I agree that snapshot tests are not sufficient and/or do not replace more specific unit tests which test the component's "critical functionality", but I believe they can still serve a purpose. For example, they can guard against unexpected DOM changes and second-order effects (e.g. when changing a component fails a snapshot tests in another package)

File: packages/edit-site/src/components/sidebar/global-styles/screen-header.js
Comment (ciampo) [inline packages/edit-site/src/components/sidebar/global-styles/screen-header.js:37]: In terms of content hierarchy, should we expose the heading's level as a prop, so that it can be changed depending on the level of nesting of the screen?

Comment (ciampo) [reply packages/edit-site/src/components/sidebar/global-styles/screen-header.js:37]: Same would apply to the `Subtitle` component in this PR

File: packages/edit-site/src/components/sidebar/global-styles/palette.js
Comment (ciampo) [inline packages/edit-site/src/components/sidebar/global-styles/palette.js:27]: It may be a good idea to move the path computation logic to a separate file for all screens. This would also avoid the risk of introducing typos in the paths

**Comment by ciampo** [comment]
> I'd love for us to use UI components directly and not have to tweak spacing at all

Could you give me some examples, so that I can better understand if/where is the current limitation with the components?

> This is advancing so fast, incredible work Riad. [...] The sliding mechanism and even the color stacks are looking great,

Absolutely! It's good to see how the time spent on introducing / refining components is starting to pay off — the `Palette` component is a good example.

File: packages/components/src/item-group/test/__snapshots__/index.js.snap
Comment (youknowriad) [reply packages/components/src/item-group/test/__snapshots__/index.js.snap:15]: You guessed it but yeah I'm not a big fan of this kind of tests. I understand the principle but in my experience, they rarely catch any error and only add maintenance burden.

File: packages/edit-site/src/components/sidebar/global-styles/screen-header.js
Comment (youknowriad) [reply packages/edit-site/src/components/sidebar/global-styles/screen-header.js:37]: > In terms of content hierarchy, should we expose the heading's level as a prop, so that it can be changed depending on the level of nesting of the screen?

No, I think that would be the wrong approach, I'd rather we normalize these across all global styles sidebar screens. All header should be the same and all subheaders should be the same. Let's see how it evolves and if we really need more levels.

File: packages/edit-site/src/components/sidebar/global-styles/palette.js
Comment (youknowriad) [reply packages/edit-site/src/components/sidebar/global-styles/palette.js:27]: I agree, I think the whole sidebar components need to be better organized, I'll do it in a separate PR.

**Comment by youknowriad** [comment]
> Could you give me some examples, so that I can better understand if/where is the current limitation with the components?

In several occasions, I had to add margin/paddings around panels (preview, palette...), I'd have loved to use some components directly like Card/CardBody. The default styles for these didn't work very well in my quick trials. That said, I'd like to explore this further later. As a consumer of components, I shouldn't be thinking about design but more about composition.

File: packages/edit-site/src/components/sidebar/global-styles/screen-header.js
Comment (ciampo) [reply packages/edit-site/src/components/sidebar/global-styles/screen-header.js:37]: Makes sense

**Comment by ciampo** [comment]
> In several occasions, I had to add margin/paddings around panels (preview, palette...), I'd have loved to use some components directly like Card/CardBody. The default styles for these didn't work very well in my quick trials. That said, I'd like to explore this further later.

I don't really know (or haven't used) the Panel component.

The `Card` component that you mentioned should offer a way of adding padding around its set of components via [the `size` prop](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/card/card#size-string), it'd be interesting to know what didn't work very well when using it.

> As a consumer of components, I shouldn't be thinking about design but more about composition.

I definitely agree with this statement in principle. I guess the "disconnect" here is that our components library is still quite "low-level" — e.g. we have individual color indicators, text, form controls... and we also have "layout" components like Flex, Grid, and the H|V|ZStacks.

We don't have many "higher-level" components that compose these lower-level components into patterns that can be reused across Gutenberg (I'm not even sure if and which higher-level patterns should live in the components package).

**Comment by youknowriad** [comment]
Anything blocking this PR? What's missing here?

**Review by ciampo — APPROVED** [review]
Before merging, could you just rebase this PR to include changes from #35152, and then remove the `isAction` prop from the `Item` component?

Otherwise, LGTM 🎉

**Review by mtias — APPROVED** [review]

**Review by ciampo — APPROVED** [review]

**Pull request merged**
